src/Sylius/Component/Core/Calculator/ProductVariantPriceCalculator.php

```diff
@@ -50,29 +43,30 @@ public function calculateOriginal(ProductVariantInterface $productVariant, array
         $channelPricing = $productVariant->getChannelPricingForChannel($context['channel']);
 
         if (null === $channelPricing) {
-            throw new MissingChannelConfigurationException(sprintf(
-                'Channel %s has no price defined for product variant %s',
-                $context['channel']->getName(),
-                $productVariant->getName()
-            ));
+            throw new MissingChannelConfigurationException($this->getMissingChannelProductVariantPriceMessage($context['channel'], $productVariant));
         }
 
         if (null === $channelPricing->getOriginalPrice()) {
-            if ($channelPricing->getPrice() === null) {
-                $message = sprintf('Channel %s has no price defined for product variant', $context['channel']->getName());
-
-                if ($productVariant->getName() !== null) {
-                    $message .= sprintf(' %s (%s)', $productVariant->getName(), $productVariant->getCode());
-                } else {
-                    $message .= sprintf(' with code %s', $productVariant->getCode());
-                }
-
-                throw new MissingChannelConfigurationException($message);
+            if ($channelPricing->getPrice() !== null) {
+                return $channelPricing->getPrice();
             }
 
-            return $channelPricing->getPrice();
+            throw new MissingChannelConfigurationException($this->getMissingChannelProductVariantPriceMessage($context['channel'], $productVariant));
         }
 
         return $channelPricing->getOriginalPrice();
     }
+
+    private function getMissingChannelProductVariantPriceMessage(ChannelInterface $channel, ProductVariantInterface $productVariant): string
```

Comment:
Can you move it to a static factory method inside the MissingChannelConfigurationException?
So it can also be used downstream in custom calculators.
It should look like this:
```php
throw new MissingChannelConfigurationException::createForProductVariantAndChannel($productVariant, $channel);
```
There's also a practice of creating separate factory classes, like `MissingChannelConfigurationExceptionFactory` for these methods, but it's probably too much for this case.

Reply:
@lruozzi9 why you didn't do it like in the example I gave?

Reply:
I see there's a mistake in my example, it should not have the `new ` keyword.

Reply:
Sorry, I misunderstood the comment. If I have to make a method that instantiates the class inside itself then I like most the idea of having an exception factory as you wrote.

Reply:
Unfortunately I have to ask you to make some changes again because of this misunderstanding.
There's also a small improvement regarding the exception message I spotted this time.

Reply:
NP, what do you think of this Factory class https://github.com/Sylius/Sylius/pull/12549/commits/089788f52e5a34352ec7a3c69403ca64e0901bcb?

Reply:
If you want to go with an exception factory, then you can just apply my suggestions first then extract the method to a factory class and change the `self` with the exception class.
In case you need some examples, you can find a lot here https://github.com/nelmio/alice/tree/master/src/Throwable/Exception.

Reply:
I couldn't reapply your comments because I had already made some changes. Now they should all be applied.